x-pack/plugins/enterprise_search/public/applications/app_search/utils/format_api_name.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export function formatApiName(rawName: string) {
+  return rawName
+    .trim()
+    .replace(/[^a-zA-Z0-9]+/g, '-')
+    .replace(/^[^a-zA-Z0-9]+/, '')
+    .replace(/[^a-zA-Z0-9]+$/, '')
```

Comment:
I know we're just copying this over from ent-search, but it's not super clear to me why those last 2 regexes couldn't be simplified to:

```js
    .replace(/^[-]+/, '')
    .replace(/[-]+$/, '')
```
since the first line takes care of transforming special characters into `-`. Or you could even do
```js
    // Strip leading and trailing dashes
    .replace(/^[-]+|[-]+$/, '')
```
just IMO, comments should be mandatory when writing regexes because they're so annoying to parse :)

Reply:
+++plus++PLUS+